pilot/pkg/model/authentication.go

```diff
@@ -59,6 +61,16 @@ type AuthenticationPolicies struct {
 
 	peerAuthentications map[string][]Config
 
+	// namespaceMutualTLSMode is the MutualTLSMode correspoinding to the namespace-level PeerAuthentication.
+	// All namespace-level policies, and only them, are added to this map. If the policy mTLS mode is set
+	// to UNSET, it will be resolved to the globalMutualTLSMode, if exist (i.e not UNKNOWN), or MTLSPermissive
+	// otherwise.
+	namespaceMutualTLSMode map[string]MutualTLSMode
+
+	// globalMutualTLSMode is the MutualTLSMode corresponding to the mesh-level PeerAuthentication.
+	// This value can be MTLSUnknown, if there is no mesh-level policy.
+	globalMutualTLSMode MutualTLSMode
```

Comment:
you don't have to duplicate this mode. we can always calculate this value on demand. by policy[rootns], no?

Reply:
Just more convenient and readable to use a dedicated variable, imo. It needs to be treated slightly different, with all the fallback and inheritance logics.

Reply:
similarly to `namespaceMutualTLSMode`.

Reply:
namespace is even more work, as it need find the one name default (and/or have no selector). It is more or less the same as my attempt before which use the applier (much simpler code, but then slightly less efficient).

Reply:
its still possible with cleaner and efficiency. if you store namespace policy separately in the `model/authentication.go`, not mingled with ones having selector in the same `peer` field. and then in the NewApplier, passing namespace configs precisely, plus an empty label selector. empty jwt policy, etc.

i think it's fine to leave as it is as release cut approaching, above is just a way to hold on to package separate